relayer/src/supervisor.rs

```diff
@@ -161,16 +235,30 @@ impl Supervisor {
                 continue;
             }
 
-            println!("[{}] events: {:#?}", chain_id, events);
+            debug!(
+                "[{}] chain {} sent {} for object {:?}, direction {:?}",
+                self,
+                chain_id,
+                VecIbcEvents(events.clone()),
+                object,
+                direction
+            );
 
             if let Some(worker) = self.worker_for_object(object, direction) {
-                worker.send_packet_events(height, events, chain_id.clone())?;
+                worker.send_events(height, events, chain_id.clone())?;
             }
         }
 
-        if collected.has_new_blocks() {
-            for worker in self.workers.values() {
-                worker.send_new_blocks(height, collected.new_blocks.clone())?;
+        if let Some(IbcEvent::NewBlock(new_block)) = collected.new_block {
+            for (object, worker) in self.workers.iter() {
+                match object {
+                    Object::UnidirectionalChannelPath(p) => {
+                        if p.src_chain_id == src_chain.id() {
+                            worker.send_new_block(height, new_block)?;
+                        }
+                    }
+                    Object::Client(_) => {}
+                }
```

Comment:
Let's add a `fn notify_new_block(&self, src_chain: &ChainId) -> bool` method on `Object` and use this to decide whether or not to send new block events?

```suggestion
            for (_, worker) in self.workers.iter().filter(|(o, _)| o.notify_new_block(src_chain.id)) {
                    worker.send_new_block(height, new_block)?;
```